proxy/src/main/java/com/wavefront/agent/listeners/tracing/SpanUtils.java

```diff
@@ -185,4 +184,13 @@ public static String toStringId(ByteString id) {
     UUID uuid = new UUID(mostSigBits, leastSigBits);
     return uuid.toString();
   }
+
+  public static void addSpanLine(Span span, SpanLogs spanLogs) {
+    String policyId = null;
+    if (span != null && span.getAnnotations() != null) {
+      policyId = AnnotationUtils.getValue(span.getAnnotations(),
+          SPAN_SAMPLING_POLICY_TAG);
+    }
+    spanLogs.setSpan(SPAN_SAMPLING_POLICY_TAG + "=" + (policyId == null ? "NONE" : policyId));
```

Comment:
Could we get an explicit unit test in `SpanUtilsTest` for this line? To see if any tests would fail I changed the line to:

`spanLogs.setSpan(SPAN_SAMPLING_POLICY_TAG + "=" + (policyId == null ? "NONE" : "FOO"));`

One test failed:

`com.wavefront.agent.listeners.tracing.JaegerTChannelCollectorHandlerTest.testJaegerDebugOverride`.

But, that seems to be a happy accident that a failure occurred, and isn't explicitly testing how `addSpanLine()` is supposed to behave.

Reply:
There is a test case hitting the case when policyId is not null. It verifies that call.
https://github.com/zen10001/wavefront-proxy/blob/master/proxy/src/test/java/com/wavefront/agent/listeners/tracing/JaegerTChannelCollectorHandlerTest.java#:~:text=_sampledByPolicy%3Dtest
Is this sufficient?

Reply:
My only nit is that our single test case for when policyId is not null exists in `JaegerTChannelCollectorHandlerTest`. If that test is deleted (say Jaeger's Thrift over UDP is deprecated/deleted one day), then we don't have coverage for the other places it's used.